File: Greedy/2808.Minimum Seconds to Equalize a Circular Array/Solution.java
class Solution {
    public int minimumSeconds(List<Integer> nums) {
        int ans = nums.size(), n = nums.size();
        Map<Integer, List<Integer>> g = new HashMap<>();
        for (int i = 0; i < n; i++) {
            g.computeIfAbsent(nums.get(i), k -> new ArrayList()).add(i);
        }
        for (List<Integer> q : g.values()) {
            int d = n - (q.get(q.size() - 1) - q.get(0) + 1);
            for (int i = 1; i < q.size(); i++) {
                d = Math.max(d, q.get(i) - q.get(i - 1) - 1);
            }
            ans = Math.min(ans, (d + 1) / 2);
        }
        return ans;
    }
}

File: Greedy/2808.Minimum Seconds to Equalize a Circular Array/Solution.py
class Solution:
    def minimumSeconds(self, nums: List[int]) -> int:
        ans = n = len(nums)
        g = defaultdict(list)
        for i, v in enumerate(nums):
            g[v].append(i)
        for q in g.values():
            d = n - (q[-1] - q[0] + 1)
            for i in range(1, len(q)):
                d = max(d, q[i] - q[i - 1] - 1)
            ans = min((d + 1) // 2, ans)
        return ans